tests/unit/parts/plugins/test_matter_sdk_plugin.py

```diff
@@ -0,0 +1,146 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import pytest
+from craft_parts import Part, PartInfo, ProjectInfo
+
+from snapcraft.parts.plugins import MatterSdkPlugin
+
+# The repository where the matter SDK resides.
+MATTER_SDK_REPO = "https://github.com/project-chip/connectedhomeip"
+
+
+@pytest.fixture(autouse=True)
+def part_info(new_dir):
+    yield PartInfo(
+        project_info=ProjectInfo(
+            application_name="test", project_name="test-snap", cache_dir=new_dir
+        ),
+        part=Part("my-part", {}),
+    )
+
+
+def test_get_pull_commands(part_info):
+    properties = MatterSdkPlugin.properties_class.unmarshal(
+        {"matter-sdk-version": "master"}
+    )
+    plugin = MatterSdkPlugin(properties=properties, part_info=part_info)
+
+    sdk_version = properties.matter_sdk_version  # type: ignore
+
+    expected_commands = []
```

Comment:
Segmenting the list in the plugin is fine since you have comments interspersed, but in the unit tests, I recommend just creating it outright:

```suggestion
    expected_commands = [
            "if [ ! -d matter ]; then",
            "    git init",
            f"   git remote add origin {MATTER_SDK_REPO}",
            f"   git fetch --depth 1 origin {sdk_version}",
            "    git checkout FETCH_HEAD",
            "fi",
            "scripts/checkout_submodules.py --shallow --platform linux",
    ]
```